src/runtime/rpc/rpc_host_port.h

```diff
@@ -57,10 +59,18 @@ class host_port
         return os << hp.to_string();
     }
 
+    rpc_group_host_port *group_host_port() const
+    {
+        CHECK_NOTNULL(_group_host_port, "group_host_port cannot be null!");
+        return _group_host_port;
+    }
+    void assign_group(const char *name);
+
 private:
     std::string _host;
     uint16_t _port = 0;
     dsn_host_type_t _type = HOST_TYPE_INVALID;
+    rpc_group_host_port *_group_host_port = nullptr;
```

Comment:
Use ref_ptr<rpc_group_host_port> to manage `ref_counter`-typed pointer ? Or just shared_ptr ?